extensions/amp-analytics/0.1/scroll-manager.js

```diff
@@ -40,14 +43,29 @@ export class ScrollManager {
    * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
    */
   constructor(ampdoc) {
+    /** @const @private {!../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampdoc_ = ampdoc;
+
     /** @const @private {!../../../src/service/viewport/viewport-interface.ViewportInterface} */
-    this.viewport_ = Services.viewportForDoc(ampdoc);
+    this.viewport_ = Services.viewportForDoc(this.ampdoc_);
```

Comment:
Does `<amp-next-page>` share the same ampdoc with the parent page? what about the viewport service?

Reply:
Replied on caroline's comment